core/kernel/tee_ta_manager.c

```diff
@@ -116,6 +116,9 @@ static bool tee_ta_try_set_busy(struct tee_ta_ctx *ctx)
 {
 	bool rc = true;
 
+	if (ctx->flags & TA_FLAG_CONCURRENT)
+		return true;
```

Comment:
This should behave has is no `single_instance_lock` is required, i.e:
```
if (ctx->busy)
	return false;
else {
	ctx->busy = true;
	return true;
}
```

Reply:
`busy` is set each time a TA is entered. With what you suggest then there would be no concurrency (actually I think we would panic due to unexpected false from `tee_ta_try_set_busy()`) or am I missing something?

Reply:
Ok, I understand. In this scope, yes, `TA_FLAG_CONCURRENT` is restricted to pTAs only.